### DISCUSSION TITLE
How to run esp32_3248s035c on arduino platform ?

### DISCUSSION THREAD

**Discussion by eagl1** [rzeldent/esp32-smartdisplay#168] [Q&A] [answered]
Hi,

I'm trying to compile and program this display on arduino ide.

I have the esp32_3248s035c 3.5" capacitive display.

Is there any source code that I can adapt for my programming ?

**Comment by rzeldent** [comment]
Hi Eagl1,

I haven't made any provisions for Arduino GUI bit think that zipping the library directory and defining the defines in the boards definition should do the trick...
I can however recommend switching to PlatformIO!

**Reply by eagl1** [reply]
That's what I'm doing now, it fails to program the board. I think it's hardware issue because it didn't program on different methods.

I tried platformio, arduino ide and flash tool.

![image](https://github.com/rzeldent/esp32-smartdisplay/assets/14331173/e78f0867-5468-4687-a380-48790a1d9725)

```
Building in release mode
Retrieving maximum program size .pio\build\esp32-8048S043C-SD\firmware.elf
Checking size .pio\build\esp32-8048S043C-SD\firmware.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM:   [===       ]  28.2% (used 92540 bytes from 327680 bytes)
Flash: [==        ]  18.0% (used 1177825 bytes from 6553600 bytes)
Configuring upload protocol...
AVAILABLE: cmsis-dap, esp-bridge, esp-builtin, esp-prog, espota, esptool, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa   
CURRENT: upload_protocol = esptool
Looking for upload port...
Auto-detected: COM6
Forcing reset using 1200bps open/close on port COM6
Waiting for the new upload port...
Uploading .pio\build\esp32-8048S043C-SD\firmware.bin
esptool.py v4.5.1
Serial port COM6
Connecting...

A serial exception error occurred: Cannot configure port, something went wrong. Original message: PermissionError(13, 'Access is denied.', None, 5)
Note: This error originates from pySerial. It is likely not a problem with esptool, but with the hardware connection or drivers.
For troubleshooting steps visit: https://docs.espressif.com/projects/esptool/en/latest/troubleshooting.html
*** [upload] Error 1
=================================== [FAILED] Took 20.87 seconds ===================================
Environment         Status    Duration
------------------  --------  ------------
esp32-8048S043C-SD  FAILED    00:00:20.869
============================== 1 failed, 0 succeeded in 00:00:20.869 ==============================
 *  The terminal process "C:\Users\eagle\.platformio\penv\Scripts\platformio.exe 'run', '--target', 'upload', '--target', 'monitor', '--environment', 'esp32-8048S043C-SD'" terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it.
```

**Reply by eagl1** [reply, marked as answer]
I switched to platformio and got it to work.

It didn't work previously, because I didn't installed it in the default install location. I should let the install procedure go as is especially with vs code.

Now it's working.

**Comment by eagl1** [comment]
Also, these are the issues I found for this display to get it to work on Arduino platform:

```
//////////////////////////////////////////////////////////
// issues for SPI display in arduino ide 
// ESP32_3248S035
1. Not programming --> install CH340 2019 driver
2. Programming but no display data 
   --> erase all flash before sketch upload
   --> enable PSRAM --> not important
```

**Comment by Starraker1** [comment]
If you are prepared to move your platform, try the brilliant Annex32 Rapid Development System which will get your module working immediately but you will have to program in BASIC.  Annex32 is only for ESP32 and ESP8266 boards but has a full implementation of BASIC.  It also allows you to program the board remotely via WiFi after the initial download by USB.
